### PR TITLE
feat(mito): preparation to implementing write

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR adds utilities for implementing write for mito2, which mainly contains conversion between protobuf types and some validations.
- Imports an experimental greptime-proto that supports row insert requests
- Adds `proto_util` mod to help convert types between db and proto
- Validates and collects write requests in `handle_write_requests`. It use `RegionWriteCtx` to batch requests and maintains sequences
- Implements some validation for `WriteRequest`.
- Implements methods to fill columns by default value for `WriteRequest`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #1869 
- Relies on #2086

### CODE REVIEW

File: src/mito2/src/proto_util.rs
Comment (waynexia) [inline src/mito2/src/proto_util.rs:1]: I think this should be placed in API crate

**Review by waynexia — APPROVED** [review]
No test so I can only choose to believe the great coder

File: src/mito2/src/proto_util.rs
Comment (evenyag) [reply src/mito2/src/proto_util.rs:1]: Now only the mito2 crate uses the new greptime-proto so I have to put it here.

File: src/mito2/src/request.rs [hunk 1]
```diff
@@ -84,9 +91,193 @@ pub struct CloseRequest {
...
+
+                return FillDefaultSnafu {
+                    region_id,
+                    column: &column.column_schema.name,
+                }
+                .fail();
+            }
+        }
+
+        // Checks all columns in rows exist in the regino.
```
Comment (killme2008) [inline src/mito2/src/request.rs:196]:
```suggestion
        // Checks all columns in rows exist in the region.
```

**Review by killme2008 — APPROVED** [review]

**Pull request merged**